src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +121,51 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(this._pollingTimer);
+        this._deinitializePollingTimer(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string) {
+        this._initializePollingTimer();
+        this._pollingInterval = window.setTimeout(async () =>  {
+            try {
+                if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status.toLowerCase() === ORDER_STATUS_APPROVED) {
```

Comment:
Would you get a status other than pending? Like rejected? Would you keep retrying?

Reply:
The expected behaviour is that we have `CREATED` status (when the payment session created) and `APPROVED` status (when the buyer confirmed this session). 
However, it is an extremely legit point for the cases when something goes not by the flow we expect! It makes absolute sense to have the second check, let's say if the status `CREATED` and not approved `APPROVED` -> we have to continue on our poller. 
If the status `CREATED` and not approved `APPROVED` -> we have to continue on our poller.
If the status is not `CREATED` and not approved `APPROVED` -> we have to reject the session because it is not something we can handle.

Thanks for the suggestion!